src/item.h

```diff
@@ -1364,7 +1367,11 @@ class item : public JsonSerializer, public JsonDeserializer, public visitable<it
         * Returns label from "item_label" itemvar and quantity
         */
         std::string label( unsigned int quantity = 0 ) const;
+
     private:
+        /** unique identifier for each non-null instance */
+        item_uid uid_;
```

Comment:
> We have recently committed code using both leading and trailing underscores.

I've been sick and busy and haven't had a chance to review much code recently, if I had a chance I would have been pushing back on this, and I will do so now.

> This is more complex though and very sensitive to JSON load order or other operations on item

Please explain the problem, AFAICS when you create a handle to an item, you can stash a uid in the item, it's not needed before a handle to the item exists, there's no ordering problem.  When the item is deserialized the uid is deserialized exactly as if it were stored any other way.  The uid is also owned by the item reference, not the item itself, so it's cleaner for the item to not even know it exists.  When a handle is released by an item_location it can clean up the uid, if it gets orphaned, it leaks the uid, no big deal.